managed/services/dbaas/kubernetes/client/client.go

```diff
@@ -49,27 +63,79 @@ var (
 	}
 )
 
+// Client is the internal client for Kubernetes.
+type Client struct {
+	clientset       kubernetes.Interface
+	dbClusterClient *database.DatabaseClusterClient
+	restConfig      *rest.Config
+	namespace       string
+}
+
 // NewFromInCluster returns a client object which uses the service account
 // kubernetes gives to pods. It's intended for clients that expect to be
 // running inside a pod running on kubernetes. It will return ErrNotInCluster
 // if called from a process not running in a kubernetes environment.
 func NewFromInCluster() (*Client, error) {
-	c, err := inClusterConfig()
+	config, err := inClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+	config.QPS = defaultQPSLimit
+	config.Burst = defaultBurstLimit
+	clientset, err := newForConfig(config)
 	if err != nil {
 		return nil, err
 	}
+	c := &Client{
+		clientset:  clientset,
+		restConfig: config,
+	}
+	err = c.setup()
+	return c, err
+}
 
-	clientset, err := newForConfig(c)
+// NewFromKubeConfigString creates a new client for the given config string.
+// It's intended for clients that expect to be running outside of a cluster
+func NewFromKubeConfigString(kubeconfig string) (*Client, error) {
+	config, err := clientcmd.BuildConfigFromKubeconfigGetter("", NewConfigGetter(kubeconfig).loadFromString)
+	if err != nil {
+		return nil, err
+	}
+	config.QPS = defaultQPSLimit
+	config.Burst = defaultBurstLimit
+	clientset, err := newForConfig(config)
 	if err != nil {
 		return nil, err
 	}
+	c := &Client{
+		clientset:  clientset,
+		restConfig: config,
+	}
+	err = c.setup()
+	return c, err
+}
 
+func (c *Client) setup() error {
 	namespace := "default"
 	if space := os.Getenv("NAMESPACE"); space != "" {
 		namespace = space
 	}
+	// Set PATH variable to make aws-iam-authenticator executable
+	path := fmt.Sprintf("%s:%s", os.Getenv("PATH"), dbaasToolPath)
+	os.Setenv("PATH", path)
```

Comment:
@gen1us2k is this needed after all?
https://github.com/percona/pmm/pull/1407#discussion_r1023703215